Tests/DatadogTests/Datadog/RUMMonitorTests.swift

```diff
@@ -551,4 +566,22 @@ class RUMMonitorTests: XCTestCase {
 
         try Datadog.deinitializeOrThrow()
     }
+
+    // MARK: - Private helpers
+
+    private var expectedAttributes = [String: String]()
+    private func setGlobalAttributes(of monitor: DDRUMMonitor) {
+        expectedAttributes = [String.mockRandom(): String.mockRandom()]
+        expectedAttributes.forEach {
+            monitor.setAttribute(forKey: $0, value: $1)
+        }
+    }
+
+    private func verifyGlobalAttributes(in matchers: [RUMEventMatcher]) {
+        for matcher in matchers {
+            expectedAttributes.forEach { attrKey, attrValue in
+                XCTAssertEqual(try? matcher.attribute(forKeyPath: attrKey), attrValue)
+            }
+        }
+    }
 }
```

Comment:
Beside all the test added to `RUMMonitorTests` (which ensure that the attributes are aggregated well in all commands 👌), we also need the behavioural test of attributes modification. After we clarify the product requirement for attributes processing, something like the following would be very handy IMO, WDYT?
```swift

// MARK: - Sending Attributes

func testSendingAttributes() throws {
    RUMFeature.instance = .mockByRecordingRUMEventMatchers(directory: temporaryDirectory)
    defer { RUMFeature.instance = nil }

    let view1 = createMockView(viewControllerClassName: "FirstViewController")
    let view2 = createMockView(viewControllerClassName: "SecondViewController")

    let monitor = RUMMonitor.initialize()

    // set global attributes:
    monitor.setAttribute(forKey: "attribute1", value: "value 1")
    monitor.setAttribute(forKey: "attribute2", value: "value 2")

    // start View 1:
    monitor.startView(viewController: view1)

    // update global attributes while the View 1 is active
    monitor.setAttribute(forKey: "attribute1", value: "changed value 1") // change the attribute value
    monitor.setAttribute(forKey: "attribute2", value: nil) // remove the attribute

    monitor.stopView(viewController: view1)

    // start View 2:
    monitor.startView(viewController: view2)
    monitor.stopView(viewController: view2)

    let rumEventMatchers = try RUMFeature.waitAndReturnRUMEventMatchers(count: 3)
    let firstViewEvent = try rumEventMatchers
        .lastRUMEvent(ofType: RUMView.self) { rumModel in rumModel.view.url == "FirstViewController" }

    XCTAssertEqual(try firstViewEvent.attribute(forKeyPath: "attribute1") as String, "changed value 1")
    XCTAssertEqual(try firstViewEvent.attribute(forKeyPath: "attribute2") as String, "value 2")

    let secondViewEvent = try rumEventMatchers
        .lastRUMEvent(ofType: RUMView.self) { rumModel in rumModel.view.url == "SecondViewController" }

    XCTAssertEqual(try secondViewEvent.attribute(forKeyPath: "attribute1") as String, "changed value 1")
    XCTAssertNil(try? secondViewEvent.attribute(forKeyPath: "attribute2") as String)
}
```